cirq/google/xmon_device.py

```diff
@@ -42,6 +42,13 @@ def __init__(self,
             exp_11_duration: The maximum duration of an ExpZ operation.
             qubits: Qubits on the device, identified by their x, y location.
         """
+        if isinstance(measurement_duration, timedelta):
```

Comment:
Move this conditional logic into `Duration.from`. Have `Duration.from` return its argument if that argument is already a duration, wrap it if it's a time delta, and otherwise raise an error.

Reply:
And that would be a TypeError?

Reply:
or a value error. Either is fine.